src/QMCApp/QMCMain.cpp

```diff
@@ -551,56 +552,64 @@ bool QMCMain::runQMC(xmlNodePtr cur, bool reuse)
   std::string prev_config_file = last_driver ? last_driver->get_root_name() : "";
   bool append_run              = false;
 
-  if (!population_)
+  // Within this scope if you split myComm you must catch UniformMPIError for your split MPI comm scope.
```

Comment:
Please move this message to UniformMPIError.
> you must catch UniformMPIError for your split MPI comm scope.

Better to say the exception must be caught at the first accessible communicator from the top of the call stack.